# Housing Market

### Introduction:

This time we will create our own dataset with fictional numbers to describe a house market. As we are going to create random data don't try to reason of the numbers.

### Step 1. Import the necessary libraries

In [40]:
import pandas as pd 
import numpy as np 

### Step 2. Create 3 differents Series, each of length 100, as follows: 
1. The first a random number from 1 to 4 
2. The second a random number from 1 to 3
3. The third a random number from 10,000 to 30,000

In [41]:
np.random.seed(123)
s1 = pd.Series(np.random.randint(1,5,size=100))
s2 = pd.Series(np.random.randint(1,4,size=100))
s3 = pd.Series(np.random.randint(10000,30000,size=100))

### Step 3. Let's create a DataFrame by joinning the Series by column

In [42]:
df = pd.concat([s1,s2,s3],axis=1)
df.head()

0  1      2
0  3  2  20802
1  2  1  23442
2  3  3  18121
3  3  2  19123
4  1  2  15650

### Step 4. Change the name of the columns to bedrs, bathrs, price_sqr_meter

In [43]:
df.rename(columns = {0:'bedrs',1:'bathrs',2:'price_sqr_meter'},inplace = True)
df

bedrs  bathrs  price_sqr_meter
0       3       2            20802
1       2       1            23442
2       3       3            18121
3       3       2            19123
4       1       2            15650
5       3       1            20715
6       3       2            22214
7       2       2            20743
8       4       1            14169
9       3       1            25313
10      4       2            29006
11      2       3            28115
12      3       2            28059
13      2       2            20332
14      1       2            29331
15      2       3            16701
16      3       1            24970
17      4       3            25547
18      2       3            29227
19      1       2            26825
20      3       3            14718
21      1       3            21771
22      4       2            15731
23      2       3            20958
24      4       1            29958
25      3       3            19528
26      2       3            25319
27      1       2            26996
28      1       2            17044
29      1       3            14421
..    ...     ...              ...
70      1       1            12621
71      4       3            23301
72      4       3            14343
73      4       2            22023
74      2       2            21278
75      1       1            12878
76      3       3            10924
77      1       3            19342
78      3       2            12191
79      2       3            18113
80      4       3            12512
81      3       1            10639
82      1       3            22782
83      3       1            13054
84      3       1            21860
85      4       2            11771
86      1       3            16895
87      4       2            15338
88      4       2            24852
89      3       1            22437
90      4       2            26928
91      2       2            25745
92      4       3            24178
93      3       1            24442
94      2       2            23778
95      4       2            28023
96      2       1            11594
97      3       3            18102
98      1       2            17341
99      1       2            29222

[100 rows x 3 columns]

### Step 5. Create a one column DataFrame with the values of the 3 Series and assign it to 'bigcolumn'

In [44]:
bigcolumn = pd.concat([s1,s2,s3],axis=0)
type(bigcolumn)

pandas.core.series.Series

In [ ]:
bigcolumn = pd.DataFrame(bigcolumn)
type(bigcolumn)

### Step 6. Ops it seems it is going only until index 99. Is it true?

In [54]:
print(bigcolumn.shape)
bigcolumn.tail()

(300, 1)


0
295  28023
296  11594
297  18102
298  17341
299  29222

### Step 7. Reindex the DataFrame so it goes from 0 to 299

In [53]:
bigcolumn = bigcolumn.reset_index(drop=True)
bigcolumn.tail()

0
295  28023
296  11594
297  18102
298  17341
299  29222